taichi/program/program.cpp

```diff
@@ -68,9 +70,19 @@ Program::Program(Arch desired_arch) : snode_rw_accessors_bank_(this) {
   if (config.debug)
     config.check_out_of_bound = true;
 
-  profiler = make_profiler(config.arch);
+  profiler = make_profiler(config.arch, config.kernel_profiler);
   llvm_program_ = std::make_unique<LlvmProgramImpl>(config, profiler.get());
 
+#if defined(TI_WITH_CUDA)
```

Comment:
Can you move this back to llvm_program.cpp? :P We plan to decouple backend-specific logic from Program class.

Reply:
Previously, I encountered a segfault with a null pointer ( `profiler` ). I will move the code back and try to solve the problem